producer/src/blockchain/blockchain.go

```diff
@@ -258,3 +258,84 @@ func GetBlockByHash(hash []byte, filename string, database string) ([]byte, erro
 
 	return bl, nil
 }
+
+/*
+Takes in the ledger file's name and the desired name for the metadata file
+Creates the database if it doesn't exist, inserts the appropriate column names
+Moves through the ledger file and extracts blocks, repopulating the metadata file
+Test will check to make sure a ledger populated by AddBlock() will result in the same
+metadata table as would result from here
+Return an error if anything I/O operations fail with detailed message
+*/
+func RecoverBlockchainMetadata(ledgerFilename string, metadataFilename string) error {
+	_, err := os.Stat(metadataFilename)
+	if err != nil {
+		// create database
+		_, err = os.Create("Database.db")
+
+		// open database
+		db, err := sql.Open("sqlite3", "testDatabase.db")
+		if err != nil {
+			return errors.New("Failed to open newly created database")
+		}
+		defer db.Close()
```

Comment:
`defer` statements usually go immediately after a successful call to `Open`. Defers get called in a stack-like order once the function returns.